Frameworks/UIKit/UIViewController.mm

```diff
@@ -1380,85 +1380,66 @@ - (void)dismissViewControllerAnimated:(BOOL)animated completion:(void (^)(void))
             position.x -= DisplayProperties::ScreenWidth();
         }
 
-        [[UIApplication sharedApplication] beginIgnoringInteractionEvents];
-        [animation setToValue:[NSValue valueWithCGPoint:position]];
-        [animation setDuration:0.2f];
-        [animation setBeginTime:CACurrentMediaTime()];
-        [animation setTimingFunction:[CAMediaTimingFunction functionWithName:@"kCAMediaTimingFunctionEaseInEaseOut"]];
-        [animation
-            setDelegate:[_TransitionNotifier _transitionTrampoline:self withSelector:@selector(_dismissTransitionStopped:finished:)]];
-        [animation setRemovedOnCompletion:NO];
-        [layer addAnimation:animation forKey:@"ModalDismiss"];
-
-        priv->_dismissCompletionBlock.attach([cleanupCompletion copy]);
-
-        priv->_dismissController = presented;
-        [presented _notifyViewWillDisappear:YES];
+        [UIView animateWithDuration:0.2f
+            delay:0.0f
+            options:UIViewAnimationOptionCurveEaseInOut
+            animations:^{
+                presentedView.center = position;
+                [presented->priv->_modalOverlayView setAlpha:0.0f];
+            }
+            completion:^(BOOL completed) {
+                [self _finishModalDismissAnimated:YES completion:completion];
+            }];
     } else {
-        [presented _notifyViewWillDisappear:animated];
-        [presentedView removeFromSuperview];
-
-        if ([[self view] superview] == nil) {
-        } else {
-            [[[self view] superview] bringSubviewToFront:[self view]];
-        }
-
-        [presented _notifyViewDidDisappear:NO];
-        [self _notifyViewDidAppear:animated];
-
-        cleanupCompletion();
+        [self _finishModalDismissAnimated:NO completion:completion];
     }
 }
 
-- (void)_notifyDidAppearAnimated:(UIView*)view {
+- (void)_notifyDidAppear:(UIView*)view animated:(BOOL)animated {
```

Comment:
This is the scariest change in that there's no test coverage (previously only the animated code path used this method to recursively call viewDidAppear on any view/subview with an owning controller (likely the existing code relied on the default handling in UIView.mm to handle this in the non-animated case). I wanted to make sure there was no incorrect behavior change after disabling the default appearance handling in UIView.mm so this method is now invoked in the non-animated case as well. Note that the underlying scenario (the need to invoke appearance events for a controller whose view is a subview of a controller being presented via presentViewController) seems obscure. Same comments apply for the similar method that handles viewDidDisappear.